src/test/java/io/vertx/core/http/HttpTest.java

```diff
@@ -5913,4 +5914,88 @@ private void testClientRequestWithLargeBodyInSmallChunks(boolean chunked) throws
     }));
     await();
   }
+
+
+  @Test
+  public void testProxyProtocolIllegalHeader() throws Exception {
+    HAProxy proxy = new HAProxy(DEFAULT_HTTP_HOST, DEFAULT_HTTP_PORT, Buffer.buffer());
+    proxy.start(vertx);
+
+    server.close();
+    server = vertx.createHttpServer(createBaseServerOptions()
+      .setUseProxyProtocol(true))
+      .exceptionHandler(exception -> {
+        if(exception instanceof io.netty.handler.codec.haproxy.HAProxyProtocolException)
+          testComplete();
+      });
+    server.requestHandler(req -> {
+      fail("Should not be called");
+    });
+
+    startServer();
+    client.request(HttpMethod.GET, proxy.getPort(), proxy.getHost(), DEFAULT_TEST_URI)
+      .onComplete(event -> {
+        assertTrue(event.failed());
```

Comment:
add a `complete()` statement after `assertTrue(event.failed())` to end the test and be sure that the assertion will be reported correctly to the test